lxd/network/acl/acl_ovn.go

```diff
@@ -1131,3 +1148,56 @@ func ovnParseLogEntry(input string, prefix string) string {
 
 	return string(out)
 }
+
+// ovnParseLogEntriesFromJournald reads the OVN log entries from the systemd journal and returns them as a list of string entries.
+func ovnParseLogEntriesFromJournald(l logger.Logger, systemdUnitName string, prefix string) ([]string, error) {
+	var logEntries []string
+	res, err := shared.RunCommand(
+		"/usr/bin/journalctl",
+		"--unit", systemdUnitName,
+		"--directory", "/var/lib/snapd/hostfs/var/log/journal",
+		"--no-pager",
+		"--boot", "0",
+		"--case-sensitive",
+		"--grep", "acl_log",
+		"--output-fields", "MESSAGE",
+		"-n", "5000",
+		"-o", "json",
+	)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read journal entries: %v", err)
+	}
+
+	lines := strings.Split(res, "\n")
```

Comment:
why are we reading up to 5000 lines of data into a variable res and then duplicating that data into a slice (up to 10k entries now), rather than consuming the line by line output of the command in a stream?

I think you should be able to use shared.RunCommandWithFds combined with either a bytes.Buffer and json.NewDecoder(&buf).Decode(), and/or bufio.NewScanner

Reply:
I don't know if that could work but `journalctl` as a `tail` mode where it will keep feeding new data as it shows up. This can also work with `-n X` if some backlog is also needed.

Reply:
we dont want a blocking tail as that isnt how the command works today, but we also dont really want to read the entire output into memory (multiple times).

Reply:
this should be fixed